rapids_build_backend/config.py

```diff
@@ -1,17 +1,21 @@
 # Copyright (c) 2024, NVIDIA CORPORATION.
 
 import os
+from typing import TYPE_CHECKING
 
 from .utils import _get_pyproject
 
+if TYPE_CHECKING:
+    from typing import Any
+
 
 class Config:
     """Manage the build configuration for the project."""
 
     # Mapping from config option to default value (None indicates that option is
     # required) and whether it may be overridden by an environment variable or a config
     # setting.
-    config_options = {
+    config_options: "dict[str, tuple[Any, bool]]" = {
```

Comment:
This hint helps `mypy` with this:

```text
rapids_build_backend/config.py:14: error: Need type annotation for "config_options"
```

And means `mypy` will help us catch issues like "this tuple has too many elements" or "the second element of the tuple isn't a `bool`".

Reply:
Let's be more explicit about what's allowed for the first element. Something like:

```suggestion
    config_types: type = str | bool | None
    config_options: "dict[str, tuple[config_types | Callable[[], config_types], bool]]" = {
```

Reply:
sure! I just pushed a commit implementing this, could you take a look?

I didn't accept your in-browser suggestion directly for a couple reasons:

* it requires an import (`typing.Callable`) that should be up under `if TYPE_CHECKING`
* the use of `: type` like that in an assignment results in errors like these:
  - `Incompatible types in assignment (expression has type "object", variable has type "type")`
* I wanted to move all of this up to `if TYPE_CHECKING`, to minimize the risk of runtime errors caused by use of this more recent syntax for type hints